### PR TITLE
migrationPrerequisite

### PR DESCRIPTION
**Pull request by Murali91**

### CODE REVIEW

File: lib/migrator.js [hunk 1]
```diff
@@ -522,4 +522,13 @@ Migrator.prototype.normalizeNames = function(names, callback) {
 };
 
 
+Migrator.prototype.migrationPrerequisite = function(callback) {
+	var self = this;
+	if (self.adapter.config && self.adapter.config.migrationTable === '_seeds') {
```
Comment (jeremygiberson) [inline lib/migrator.js]: This is introducing tight coupling between the east library and its adapters which is not desirable.

What you should do here instead is defer to the adapter to make decisions. 
This should be `if(self.adapter.migrationPrerequisite) { self.adapter.migrationPrerequisite(callback) }`

File: lib/migrator.js [hunk 2]
```diff
@@ -522,4 +522,13 @@ Migrator.prototype.normalizeNames = function(names, callback) {
 };
 
 
+Migrator.prototype.migrationPrerequisite = function(callback) {
```
Comment (jeremygiberson) [inline lib/migrator.js]: a better name for this would be `beforeMigration`

File: lib/migrator.js [hunk 3]
```diff
@@ -522,4 +522,12 @@ Migrator.prototype.normalizeNames = function(names, callback) {
 };
 
 
+Migrator.prototype.beforeMigration = function(callback) {
+	if (this.adapter.beforeMigration) {
+		this.adapter.beforeMigration(callback);
+	}
+	callback();
```
Comment (jeremygiberson) [inline lib/migrator.js]: in an else statement so it doesnt get called if adapter.beforeMigration is defined

File: test/migrator.js [hunk 4]
```diff
@@ -575,4 +581,28 @@ describe('migrator', function() {
 		});
 
 	});
+
+	describe('call beforeMigration method of adapter if available', function() {
+		it('call beforeMigration method of adapter', function(done) {
+			var callback = sinon.spy();
```
Comment (jeremygiberson) [inline test/migrator.js]: add expectation on callback for 'calledOnce'

Comment (Murali91) [reply test/migrator.js]: Here expectation on callback fails because callback() isn't getting called in migrator but in adapter so I haven't added it.

**Pull request merged**